staging/src/k8s.io/apimachinery/pkg/util/intstr/intstr.go

```diff
@@ -174,12 +174,17 @@ func getIntOrPercentValue(intOrStr *IntOrString) (int, bool, error) {
 	case Int:
 		return intOrStr.IntValue(), false, nil
 	case String:
-		s := strings.Replace(intOrStr.StrVal, "%", "", -1)
+		isPercent := false
+		s := intOrStr.StrVal
+		if strings.HasSuffix(s, "%") {
+			isPercent = true
+			s = strings.TrimSuffix(intOrStr.StrVal, "%")
+		}
 		v, err := strconv.Atoi(s)
 		if err != nil {
-			return 0, false, fmt.Errorf("invalid value %q: %v", intOrStr.StrVal, err)
+			return 0, isPercent, fmt.Errorf("invalid value %q: %v", intOrStr.StrVal, err)
```

Comment:
```suggestion
			return 0, false, fmt.Errorf("invalid value %q: %v", intOrStr.StrVal, err)
```

return values other than the error are not intended to be meaningful